pkg/controllers/dnspolicy/dnspolicy_healthchecks.go

```diff
@@ -94,70 +88,67 @@ func (r *DNSPolicyReconciler) deleteUnexpectedGatewayProbes(ctx context.Context,
 	return nil
 }
 
-func (r *DNSPolicyReconciler) expectedProbesForGateway(ctx context.Context, gw common.GatewayWrapper, dnsPolicy *v1alpha1.DNSPolicy) ([]*v1alpha1.DNSHealthCheckProbe, error) {
+func (r *DNSPolicyReconciler) expectedProbesForGateway(ctx context.Context, gw common.GatewayWrapper, dnsPolicy *v1alpha1.DNSPolicy) []*v1alpha1.DNSHealthCheckProbe {
 	log := crlog.FromContext(ctx)
 	var healthChecks []*v1alpha1.DNSHealthCheckProbe
 	if dnsPolicy.Spec.HealthCheck == nil {
 		log.V(3).Info("DNS Policy has no defined health check")
-		return nil, nil
+		return healthChecks
 	}
 
-	interval := metav1.Duration{Duration: 60 * time.Second}
-	if dnsPolicy.Spec.HealthCheck.Interval != nil {
-		interval = *dnsPolicy.Spec.HealthCheck.Interval
-	}
+	for _, address := range gw.Status.Addresses {
+		matches := strings.Split(address.Value, "/")
+		if len(matches) != 2 {
+			log.V(3).Info(fmt.Sprintf("%s cannot be processed. expected <clsutername>/<hostname/ipaddress>", address.Value))
+			return nil
+		}
 
-	for _, listener := range gw.Spec.Listeners {
+		interval := metav1.Duration{Duration: 60 * time.Second}
```

Comment:
It looks good, but I'd just move that block that defines `interval` outside of the loop, otherwise we're redefining the variable in every iteration. Wouldn't be surprised that the compiler is smart enough to optimise it so it doesn't make any difference but just in case

Reply:
Done cc @mikenairn